app/pages/Page.js

```diff
@@ -4,32 +4,128 @@ import * as React from 'react';
 import type { CustomTranslatorProps } from 'types/translator';
 import { translate } from 'react-i18next';
 import { Grid } from 'semantic-ui-react';
+import { connect } from 'react-redux';
+import { Redirect, Route, withRouter } from 'react-router-dom';
+import type { Match } from 'react-router-dom';
+import type { State } from 'types/state';
 
 import NavigationBar from 'core-components/navigation/NavigationBar';
+import authentication from 'modules/authentication';
+import sidebars from 'modules/sidebars';
+
+const { isAuthenticated } = authentication.selectors;
+
+const { SidebarMenu, SidebarWrapper } = sidebars.components;
+const { getAllActiveSidebars } = sidebars.selectors;
+const { SIDEBAR_LENGTH, AMOUNT_OF_COLS_IN_GRID } = sidebars.constants;
+
+type StateProps = {
+  authenticated: boolean,
+  amountOfSidebars: number,
+};
 
 type PassedProps = {
+  needsAuth: boolean,
+  needsSidebar: boolean,
   children: React.Node,
 };
 
-type Props = CustomTranslatorProps & PassedProps;
+type SidebarProps = {
+  match: Match,
+  amountOfCols: number,
+};
+
+type Props = CustomTranslatorProps & PassedProps & StateProps & SidebarProps;
+
+// TODO: some flowTyped error from ContextRouter in react-router-dom package
+// $FlowFixMe
+const mapStateToProps = (state: State, props: PassedProps): StateProps => {
+  const {
+    needsAuth,
+    needsSidebar,
+  } = props;
+
+  let amountOfSidebars:number = 0;
+
+  if (needsSidebar) {
+    const activeSidebars = getAllActiveSidebars(state);
+    amountOfSidebars = activeSidebars != null ? activeSidebars.length : 0;
+  }
+
+  return {
+    authenticated: needsAuth ? isAuthenticated(state) : true,
+    amountOfSidebars,
+  };
+};
+
+
+const SidebarComponent = (props: SidebarProps): React.Node => {
+  const {
+    match,
+    amountOfCols,
+  } = props;
+
+  const topicId = match.params.id;
+
+  if (topicId == null) { // Null check necessary for flow
+    return null;
+  }
+
+  return (
+    <React.Fragment>
+      { amountOfCols > 0 &&
+        <Grid.Column className="sidebarWrapper" width={amountOfCols}>
```

Comment:
Why does this component have a class `sidebarWrapper`? Isn't this the domain of the `<SidebarWrapper>` component?

Reply:
The naming wasn't great indeed. It's supposed to be styling for the column that contains the sidebarWrapper